sys/xtimer/xtimer.c

```diff
@@ -263,3 +263,27 @@ void xtimer_set_timeout_flag(xtimer_t *t, uint32_t timeout)
     xtimer_set(t, timeout);
 }
 #endif
+
+uint64_t xtimer_left_usec(const xtimer_t *timer)
+{
+    unsigned state = irq_disable();
+    /* ensure we're working on valid data by making a local copy of timer */
+    xtimer_t t = *timer;
+    uint64_t now_us = xtimer_now_usec64();
+    irq_restore(state);
+
+    if (t.offset == 0 && t.long_offset == 0 &&
+            t.start_time == 0 && t.long_start_time == 0) {
+        return 0;
+    }
+    uint64_t start_us = _xtimer_usec_from_ticks64(
+        ((uint64_t)t.long_start_time << 32) | t.start_time);
+    uint64_t target_us = start_us + _xtimer_usec_from_ticks64(
+        ((uint64_t)t.long_offset) << 32 | t.offset);
+
+    if (now_us > target_us) {
```

Comment:
Just thinking loud: In the recent xtimer rework a lot of effort was put into not comparing absolute times because of overflow troubles. I can't think of a realistic scenario where the 64-bit overflow is about to happen soon enough to really care (i.e. starting from a high offset or having an insanely fast timer), but in that case this won't work as expected.

Reply:
True. Let's assume that 64bit won't overflow anytime soon. There'd be >580 __years__ when counting __nanoseconds__. With microseconds, there are __580000 years__ of space in 2**64...

Reply:
Maybe add a comment here to clarify this for the next person browsing the code.

Reply:
makes sense, done